### PR TITLE
Make `reportTests` more strict

### PR DESCRIPTION
**Pull request by marcomorain**
- Extract `resultMessage` into a function and make sure that it is only
  called once per test result.
- Replace some `any` types with `unknown` to make the intent more clear.
- Remove the `log` boolean, since this is always true.

## My Calva PR Checklist

I have:

- [x] Read [How to Contribute](https://github.com/BetterThanTomorrow/calva/wiki/How-to-Contribute#before-sending-pull-requests).
- [x] Directed this pull request at the `dev` branch. (Or have specific reasons to target some other branch.)
- [x] Made sure I have changed the PR base branch, so that it is not `published`. (Sorry for the nagging.)
- [x] Updated the `[Unreleased]` entry in `CHANGELOG.md`, linking the issue(s) that the PR is addressing.
- [x] Figured if **anything** about the fix warrants tests on Mac/Linux/Windows/Remote/Whatever, and either tested it there if so, or mentioned it in the PR.
- [x] Added to or updated docs in this branch, if appropriate
- [x] Tested the VSIX built from the PR (so, after you've submitted the PR). You'll find the artifacts by clicking _Show all checks_ in the CI section of the PR page, and then _Details_ on the `ci/circleci: build` test. NB: *You need to sign in/up at Circle CI to find the _Artifacts_ tab.*
     - [x] Tested the particular change
     - [x] Figured if the change might have some side effects and tested those as well.
     - [x] Smoke tested the extension as such.
- [x] Referenced the issue I am fixing/addressing _in a commit message for the pull request_.
     - [x] If I am fixing the issue, I have used [GitHub's fixes/closes syntax](https://help.github.com/en/articles/closing-issues-using-keywords)
     - [x] If I am fixing just part of the issue, I have just referenced it w/o any of the "fixes” keywords.
- [x] Created the issue I am fixing/addressing, if it was not present.

Ping @pez, @bpringe

### CODE REVIEW

**Review by bpringe — APPROVED** [review]
Thanks! Looks good to me, and I tested the branch in the development host. I'll let @PEZ have a look before we merge, even though it looks like nothing functionally changed.

**Comment by bpringe** [comment]
I tested the VSIX as well, at least just by running each test command in a simple test file and looking at the results. 👍

**Comment by PEZ** [comment]
This is fantastic. Thank you! 🙏

**Pull request merged**
